app/services/status_message_creation_service.rb

```diff
@@ -54,13 +58,20 @@ def add_photos(status_message, photos)
     end
   end
 
-  def process(status_message, aspect_ids, services)
-    add_to_streams(status_message, aspect_ids) unless status_message.public
+  def validate_aspects
+    raise BadAspectsIDs if aspects.empty?
+  end
+
+  def load_aspects_from_param(aspect_ids)
```

Comment:
Since you already parse the `params` outside of this method I would just call this `def load_aspects(aspect_ids)`.